packages/flutter/test/material/flexible_space_bar_test.dart

```diff
@@ -117,6 +119,67 @@ void main() {
     expect(clipRect.size.height, minExtent);
   });
 
+  // This is a regression test for https://github.com/flutter/flutter/issues/14227
+  testWidgets('FlexibleSpaceBar sets width constraints for the title', (WidgetTester tester) async {
+    const double width = 300;
+    const double height = 100;
+    const double minExtent = 100;
+    const double maxExtent = 200;
+
+    final FlexibleSpaceBarSettings customSettings = FlexibleSpaceBar.createSettings(
+      currentExtent: maxExtent,
+      minExtent: minExtent,
+      maxExtent: maxExtent,
+      child: AppBar(
+        flexibleSpace: FlexibleSpaceBar(
+          titlePadding: EdgeInsets.zero,
+          title: Container(
+            height: height,
+            child: Text(
+              'X' * 2000,
+              maxLines: 1,
+              overflow: TextOverflow.ellipsis,
+            ),
+          ),
+        ),
+      ),
+    ) as FlexibleSpaceBarSettings;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Container(
+            width: width,
+            child: CustomScrollView(
+              primary: true,
+              slivers: <Widget>[
+                SliverPersistentHeader(
+                  floating: true,
+                  pinned: true,
+                  delegate: TestDelegate(settings: customSettings),
+                ),
+                SliverToBoxAdapter(
+                  child: Container(
+                    height: 1200.0,
+                    color: Colors.orange[400],
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // The title is scaled and transformed to be 1.5 times bigger, when the
```

Comment:
Why not just verify that `tester.getRect(find.byType(Text))` is what you'd expect?

Reply:
Good idea, that made it simpler.